### PR TITLE
Add popover with column function help text

### LINKED ISSUE #12075: Add function helper popover to notebook expressions
When I start writing a function, I should see a popover with documentation of that function to help me know how to use it. It could also include a link to the correct page on metabase.com/docs

image

### PR DESCRIPTION
**Pull request by paulrosenzweig**
Resolves #12075

help text popover

### CODE REVIEW

File: frontend/src/metabase/components/Popover.css [hunk 1]
```diff
@@ -20,7 +20,7 @@
   box-shadow: 0 4px 10px var(--color-shadow);
   background-color: var(--color-bg-white);
   border-radius: 6px;
-  /* overflow: hidden; */
+  overflow: hidden;
```
Comment (paulrosenzweig) [inline frontend/src/metabase/components/Popover.css]: @tlrobinson it looks like you commented this out in #10497. Without that the yellow at the top of the popover was creating sharp corners. Any recollection what might break by removing this fence?

File: frontend/test/metabase/lib/expressions/suggest.unit.spec.js [hunk 2]
```diff
@@ -283,7 +317,7 @@ describe("metabase/lib/expression/suggest", () => {
         ).toEqual([...AGGREGATION_FUNCTIONS, ...METRICS_ORDERS, OPEN_PAREN]);
       });
 
-      it("should suggest expression operators after aggregation argument", () => {
+      xit("should suggest expression operators after aggregation argument", () => {
```
Comment (paulrosenzweig) [inline frontend/test/metabase/lib/expressions/suggest.unit.spec.js]: @tlrobinson I'm not sure why this started failing. It looks like it still works in the UI, but this test passes `startRule: "aggregation"` whereas I can only get the UI to pass `"expression"`.

**Review by kdoh — CHANGES_REQUESTED** [review]
Overall this is looking pretty good. There are just a couple things we'll want to tackle to make sure things hang together a bit more visually.

- It'd be nice if the popover width was constant and matched the width of the input box, right now there's a bit of jumpiness between the help text and the column text as you go from function to function.
-  We should increase the spacing between the labels and text a bit, right now things are kind of tight and that combined with the font weight is making things feel chunky in the wrong way.
- Could use a bit more vertical padding on the `Learn more` link so it doesn't feel so tight at the bottom.
-  I'd bump up the font size on the example, arguably that's the most useful bit of info in the popover and it's the hardest to read right now.

One thing I'd be interested in is seeing how things feel if we get rid of the labels in the first section (`What this does` and `Example`) and see if it's still easy enough to understand what a given function is for just from that content itself. They feel a bit overly important at least in their current form and make it harder to find the relevant bits about usage. Might be worth a quick try.

File: frontend/test/metabase/lib/expressions/suggest.unit.spec.js (hunk 2 above)
Comment (tlrobinson) [reply frontend/test/metabase/lib/expressions/suggest.unit.spec.js]: Expressions can be used in custom fields (startRule: "expression"), custom aggregations (startRule: "aggregation"), and custom filters (startRule: "boolean").

Add a summarize step then add a custom aggregation.

File: frontend/src/metabase/components/Popover.css (hunk 1 above)
Comment (tlrobinson) [reply frontend/src/metabase/components/Popover.css]: I don't recall, but if you uncomment it and none of the issues in #10497 regress then go for it.

**Comment by paulrosenzweig** [comment]
Updated style:

image

**Review by kdoh — APPROVED** [review]

**Pull request merged**